### PR TITLE
[Feature] Implement recommendation/trending contents

### PR DESCRIPTION
**Pull request by arclic**
## Feature
 - Implement recommendation contents
 - Implement trending contents
스크린샷 2021-11-19 오후 9 02 07

## Related Issues
#46

### CODE REVIEW

File: backend/subroker/views.py [hunk 1]
```diff
@@ -336,7 +361,56 @@ def content_detail(request, content_id):
...
-    return HttpResponse("recommendation")
+    if request.method == "GET":
+        if not request.user.is_authenticated:
+            #ERR 401 : Not Authenticated
+            return HttpResponse(status=401)
+
+        try:
+            user = User.objects.get(id=user_id)
+        except User.DoesNotExist:
+            #ERR 404 : Group Doesn't Exist
```
Comment (cw10108) [inline backend/subroker/views.py]: # User

File: backend/subroker/views.py [hunk 2]
```diff
@@ -336,7 +361,56 @@ def content_detail(request, content_id):
...
+            return HttpResponse(status=404)
+
+        fav_contents_id = [content["id"] for content in user.favorite_contents.all()]
+        recommendation_contents = []
+        default_url = 'https://api.themoviedb.org/3/movie/{0}/recommendations'
```
Comment (cw10108) [inline backend/subroker/views.py]: I think recommendation_url would be better than default_url

File: backend/subroker/views.py [hunk 3]
```diff
@@ -408,7 +482,27 @@ def content_favorite(request, user_id, content_id):
...
 
 def content_trending(request):
-    return HttpResponse("trending")
+    if request.method == "GET":
+        if not request.user.is_authenticated:
+            #ERR 401 : Not Authenticated
+            return HttpResponse(status=401)
+
+        placeholder = 'https://via.placeholder.com/150?text=No+Content'
+        url = 'https://api.themoviedb.org/3/movie/popular'
```
Comment (cw10108) [inline backend/subroker/views.py]: same as above, popular_url or trending_url would be better, I think

File: backend/subroker/views.py [hunk 4]
```diff
@@ -336,7 +361,56 @@ def content_detail(request, content_id):
...
 
 def content_recommendation(request, user_id):
-    return HttpResponse("recommendation")
+    if request.method == "GET":
+        if not request.user.is_authenticated:
+            #ERR 401 : Not Authenticated
+            return HttpResponse(status=401)
+
+        try:
+            user = User.objects.get(id=user_id)
```
Comment (cw10108) [inline backend/subroker/views.py:370]: why do we need user_id here? what about just using request.user.id?

**Review by cw10108 — COMMENTED** [review]
commented on some variable name, and parameter usage for content_recommendation

**Review by muhwagwa — APPROVED** [review]

File: backend/subroker/views.py (hunk 4 above)
Comment (arclic) [reply backend/subroker/views.py:370]: @cw10108 I think we don't need user_id in this project, but for expansion, it is better to separate.

**Pull request merged**